src/Workspaces/Core/Portable/Utilities/EditDistance.cs

```diff
@@ -47,572 +68,581 @@ private static char[] ConvertToLowercaseArray(string text)
             {
-                // here we are setting the initial 'left' array:
-
-                //      i<-|
-                //    j    |
-                //    ^   0|
-                //    -----+ 
-                //    0 X 1|
-                //    1 m 2|
-                //    2 l 3|
-                //    3 R 4|
-                //    4 e 5|
-                //    5 a 6|
-                //    6 d 7|
-                //    7 e 8|
-                //    8 r 9|
-                //
-                // This is the cost to go from an empty string to the longString, and is essentially
-                // the cost of inserting all the characters.  By doing this, we have access to 'left'
-                // values as we actualy walk through each column.
-                for (var i = 0; i < longLength; i++)
-                {
-                    costArray[i] = i + 1;
-                }
+                return targetLength <= threshold ? targetLength : BeyondThreshold;
+            }
 
-                // We only need to bother even checking the threshold if it's lower than the 
-                // cost of just creating all of t.
-                var checkThreshold = costThreshold < longLength;
+            // The is the minimum number of edits we'd have to make.  i.e. if  'source' and 
+            // 'target' are the same length, then we might not need to make any edits.  However,
+            // if target has length 10 and source has length 7, then we're going to have to
+            // make at least 3 edits no matter what.
+            var minimumEditCount = targetLength - sourceLength;
+            Debug.Assert(minimumEditCount >= 0);
 
-                // Some complicated indices here.  We effectively only want to walk the portion
-                // of the matrix that is 'offset' around the diagonal.  So as we're walking
-                // we check if we're getting past the offset, and if so, we bump our indices
-                // to skip the values we don't need to check.
+            // If the number of edits we'd have to perform is greater than our threshold, then
+            // there's no point in even continuing.
+            if (minimumEditCount > threshold)
+            {
+                return BeyondThreshold;
+            }
 
-                var offset = costThreshold - minimumEditCount;
+            // Say we want to find the edit distance between "sunday" and "saturday".  Our initial
+            // matrix will be.
+            //
+            //           s u n d a y
+            //      ----------------
+            //      |∞ ∞ ∞ ∞ ∞ ∞ ∞ ∞
+            //      |∞ 0 1 2 3 4 5 6
+            //    s |∞ 1 
+            //    a |∞ 2
+            //    t |∞ 3
+            //    u |∞ 4
+            //    r |∞ 5
+            //    d |∞ 6
+            //    a |∞ 7
+            //    y |∞ 8
+            //
+            // Note that the matrix will always be square, or a rectangle that is taller htan it is 
```

Comment:
In this example, they have a shared prefix/suffix. Maybe "Sunday" and "Tuesdays" if you want the words to stay the same length as this example but not have a common prefix/suffix.